File: fsharp/bob/Bob.fs
module Bob

let isQuestion (s:string) : bool = 
    if s.Trim() = "" then
        false
    else 
        s.Trim() |> Seq.last = '?'

let isUpperCase s : bool = 
    if Seq.where System.Char.IsLetter s |> Seq.length = 0 then 
        false 
    else 
        String.map System.Char.ToUpper s = s


let response (input: string): string = 
    let yelled : bool = isUpperCase input
    let question : bool = isQuestion input
    if input.Trim() = "" then 
        "Fine. Be that way!"
    elif yelled && question then
        "Calm down, I know what I'm doing!"
    elif question then 
        "Sure."
    elif yelled then 
        "Whoa, chill out!"
    else 
        "Whatever."

File: csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{
    private static bool IsQuestion(string s) => s.Length == 0 ? false : s.Last() == '?';
    private static bool IsYelled(string s) => ContainsLetters(s) ? s.ToUpper() == s : false;
    private static bool ContainsLetters(string s) => s.Any(c => Char.IsLetter(c));
    public static string Response(string statement)
    {
        string s = statement.Trim();
        if (s == "") return "Fine. Be that way!";
        else return (IsQuestion(s), IsYelled(s)) switch
        {
            (true, true) => "Calm down, I know what I'm doing!",
            (true,   _ ) => "Sure.",
            (  _ , true) => "Whoa, chill out!",
            (  _ ,   _ ) => "Whatever."
        };
    }
}